subsys/shell/backends/shell_rpmsg.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2024 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/shell/shell_rpmsg.h>
+
+SHELL_RPMSG_DEFINE(shell_transport_rpmsg);
+SHELL_DEFINE(shell_rpmsg, CONFIG_SHELL_PROMPT_RPMSG, &shell_transport_rpmsg,
+	     CONFIG_SHELL_BACKEND_RPMSG_LOG_MESSAGE_QUEUE_SIZE,
+	     CONFIG_SHELL_BACKEND_RPMSG_LOG_MESSAGE_QUEUE_TIMEOUT, SHELL_FLAG_OLF_CRLF);
+
+static int rpmsg_shell_cb(struct rpmsg_endpoint *ept, void *data,
+			  size_t len, uint32_t src, void *priv)
+{
+	const struct shell_transport *transport = (const struct shell_transport *)priv;
+	struct shell_rpmsg *sh_rpmsg = (struct shell_rpmsg *)transport->ctx;
+	struct shell_rpmsg_rx rx;
+
+	if (len == 0) {
+		return RPMSG_ERR_NO_BUFF;
+	}
+
+	rx.data = data;
+	rx.len = len;
+	if (k_msgq_put(&sh_rpmsg->rx_q, &rx, K_NO_WAIT) != 0) {
```

Comment:
You provide `&rx` that points to a local variable, that seems not correct.

Reply:
`k_msg_put` will copy the data.

Reply:
Got it, the size for the copy is provided in `k_msgq_init`